File: tests/parsers/samples/code/cs/complex.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Samples {
    public class ComplexMetrics {
        public double Scale { get; }

        public ComplexMetrics(double scale = 1.0) {
            Scale = scale;
        }

        public double Distance(double a, double b) {
            return Math.Abs(a - b) * Scale;
        }

        public double RootMeanSquare(IEnumerable<double> values) {
            var list = values.ToList();
            if (list.Count == 0) {
                return 0;
            }
            var total = list.Sum(v => v * v);
            return Math.Sqrt(total / list.Count);
        }

        public static ComplexMetrics BuildDefault() {
            return new ComplexMetrics(0.85);
        }
    }
}

File: tests/parsers/samples/code/go/complex.go
package samples

import "math"

type ComplexMetrics struct {
	Scale float64
}

func NewComplexMetrics() ComplexMetrics {
	return ComplexMetrics{Scale: 0.85}
}

func (c ComplexMetrics) Distance(a, b int) float64 {
	return math.Abs(float64(a-b)) * c.Scale
}

func (c ComplexMetrics) RootMeanSquare(values []float64) float64 {
	if len(values) == 0 {
		return 0
	}
	total := 0.0
	for _, v := range values {
		total += v * v
	}
	return math.Sqrt(total / float64(len(values)))
}
